cluster-api/providers/azure/go.mod

```diff
@@ -104,52 +97,49 @@ require (
+	golang.org/x/oauth2 v0.18.0 // indirect
+	golang.org/x/sync v0.7.0 // indirect
+	golang.org/x/sys v0.21.0 // indirect
+	golang.org/x/term v0.21.0 // indirect
+	golang.org/x/text v0.16.0 // indirect
 	golang.org/x/time v0.5.0 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.4.0 // indirect
 	google.golang.org/appengine v1.6.8 // indirect
-	google.golang.org/genproto v0.0.0-20231212172506-995d672761c0 // indirect
 	google.golang.org/genproto/googleapis/api v0.0.0-20240102182953-50ed04b92917 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20240102182953-50ed04b92917 // indirect
 	google.golang.org/grpc v1.61.1 // indirect
 	google.golang.org/protobuf v1.33.0 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
-	gopkg.in/natefinch/lumberjack.v2 v2.2.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	k8s.io/api v0.28.4 // indirect
-	k8s.io/apiextensions-apiserver v0.28.4 // indirect
-	k8s.io/apimachinery v0.28.4 // indirect
-	k8s.io/apiserver v0.28.4 // indirect
-	k8s.io/client-go v0.28.4 // indirect
-	k8s.io/cloud-provider v0.28.4 // indirect
-	k8s.io/cluster-bootstrap v0.28.4 // indirect
-	k8s.io/component-base v0.28.4 // indirect
-	k8s.io/component-helpers v0.28.4 // indirect
-	k8s.io/klog/v2 v2.110.1 // indirect
-	k8s.io/kms v0.28.4 // indirect
-	k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00 // indirect
-	k8s.io/utils v0.0.0-20230726121419-3b25d923346b // indirect
-	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.1.2 // indirect
-	sigs.k8s.io/cloud-provider-azure v1.28.4 // indirect
-	sigs.k8s.io/cluster-api v1.6.3 // indirect
-	sigs.k8s.io/controller-runtime v0.16.5 // indirect
+	k8s.io/api v0.29.3 // indirect
+	k8s.io/apiextensions-apiserver v0.29.3 // indirect
+	k8s.io/apimachinery v0.29.3 // indirect
+	k8s.io/apiserver v0.29.3 // indirect
+	k8s.io/client-go v0.29.3 // indirect
+	k8s.io/cloud-provider v0.29.3 // indirect
+	k8s.io/cluster-bootstrap v0.29.3 // indirect
+	k8s.io/component-base v0.29.3 // indirect
+	k8s.io/component-helpers v0.29.3 // indirect
+	k8s.io/klog/v2 v2.120.1 // indirect
+	k8s.io/kube-openapi v0.0.0-20240117194847-208609032b15 // indirect
+	k8s.io/utils v0.0.0-20240102154912-e7106e64919e // indirect
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.28.0 // indirect
+	sigs.k8s.io/cloud-provider-azure v1.29.3 // indirect
+	sigs.k8s.io/cloud-provider-azure/pkg/azclient v0.0.2 // indirect
+	sigs.k8s.io/cloud-provider-azure/pkg/azclient/configloader v0.0.1 // indirect
+	sigs.k8s.io/cluster-api v1.7.3 // indirect
+	sigs.k8s.io/controller-runtime v0.17.3 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
-	sigs.k8s.io/structured-merge-diff/v4 v4.3.0 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.6.3
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.7.3
```

Comment:
To be safe, this should never be higher than the version of capi we're building (1.7.0 as of today).

Reply:
Interesting. And I see @sadasu is fixing by bumping the CAPI controller in 4b4dd048dafc9fd204cb50fc758fbfc7d213cbf0 :+1:

Reply:
Updating CAPI version everywhere else too. (see later commits)

Reply:
If CAPZ uses a new feature in 1.7.3, then running it against an older capi could cause runtime failures. I think what you're doing now is the correct approach. Let's just make sure to trigger jobs for all other providers and make sure they work with capi 1.7.3.